src/App.vue

```diff
@@ -35,6 +39,14 @@ export default {
       const fromDepth = from.path.split('/').length
       this.transitionName = toDepth > fromDepth ? 'slide-right' : 'slide-left'
     }
+  },
+  created(){
+    auth.onAuthStateChanged(function(user) {
+      if (user) {
+        // Add user to vuex store
+        store.commit('addUser', user)
```

Comment:
Maybe I can dispatch once here instead and do both in one function call to keep things together

Reply:
Why cant we just use a ```.then``` and change the login button to a css loader while it calls (https://loading.io/css/) feel like  that is  pretty common app  behavior as long  as it doesn't take more than like 5 seconds

Reply:
Ooo yea that's a solid idea I'll try to do a little thing there